### PR TITLE
feat: automod and statswatch integration

### PR DESCRIPTION
**Pull request by anukulpandey**
integrated _statswatch_ and _automod_ , will listen to the events on `default_network` from `.env`

### CODE REVIEW

File: lib/statswatch/constants.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+    config: {
+      testnet: {
+        marketplaceContractAddress: '0xB4630116Dca95650C1E56F3dD39c7edeb1075B38',
+        nftContractAddress: '0xc2F3BE4636A0a1ddf3b4D63ef22014DD41114336',
+        explorerGraphqlUrl: 'https://squid.subsquid.io/reef-explorer-testnet/graphql',
+        marketplaceGraphqlUrl: 'https://squid.subsquid.io/sqwid-marketplace-testnet/graphql',
```
Comment (matjazonline) [inline lib/statswatch/constants.js]: merge constants for app in one file

**Review by matjazonline — CHANGES_REQUESTED** [review]

File: lib/automod/constants.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+module.exports = {
+    clients: {
+        testnet: {
+            wsurl: 'wss://testnet.reefscan.com/graphql',
+            abi: [
+                'event ItemCreated(uint256 indexed itemId,address indexed nftContract,uint256 indexed tokenId,address creator);',
+                'event PositionUpdate(uint256 indexed positionId,uint256 indexed itemId,address indexed owner,uint256 amount,uint256 price,uint256 marketFee,uint8 state);'
+            ],
+            query: gql`
+            subscription evmEvents ($where: evm_event_bool_exp) {
```
Comment (matjazonline) [inline lib/automod/constants.js]: replace subscriptions with util-lib

**Review by matjazonline — CHANGES_REQUESTED** [review]

File: lib/automod/graphql.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const { execute } = require ('apollo-link');
+const { WebSocketLink } = require ('apollo-link-ws');
+const { SubscriptionClient } = require ('subscriptions-transport-ws');
+const ws = require ('ws');
+
+// oh i hate this
+// process.env ['NODE_TLS_REJECT_UNAUTHORIZED'] = 0;
+
+const createSubscriptionObservable = (wsurl, query, variables) => {
```
Comment (matjazonline) [inline lib/automod/graphql.js]: remove gql subscriptions from calees - use util-lib

**Review by matjazonline — CHANGES_REQUESTED** [review]

File: lib/automod/index.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    }
+    doCheckDocs ();
+}
+
+doCheckDocs ();
+
+let lastUpdated = Date.now ();
+
+
+const initAutomod = ()=>{
```
Comment (matjazonline) [inline lib/automod/index.js:180]: we will deploy separate mainnet/testnet servers so there will be 2 automod instances running - only enable automod in mainnet

**Review by matjazonline — CHANGES_REQUESTED** [review]

File: lib/automod/index.js (hunk 4 above)
Comment (anukulpandey) [reply lib/automod/index.js:180]: did it in index.js inside root of project 
`if(defaultNetwork==='reef_mainnet') {
	initAutomod();
}
`

**Review by matjazonline — APPROVED** [review]

**Pull request merged**